fix(ci): Ensure idempotence of user creation

sethp opened the pull request:
Previously, re-running `run.sh` in the same container would fail at the useradd step, because the user already exists. Instead, change that step to "create if not exists" semantics to ease interactive debugging of CI failures.

Split out from https://github.com/rust-lang/rust/pull/111891 per request by @jackh726

sethp committed:
fix(ci): Ensure idempotence of user creation

Previously, re-running `run.sh` in the same container would fail at the useradd step, because the user already exists. Instead, change that step to "create if not exists" semantics to ease interactive debugging of CI failures.

Split out from https://github.com/rust-lang/rust/pull/111891 per request by @jackh726

Mark-Simulacrum commented:
@bors r+ rollup=iffy

bors commented:
:pushpin: Commit 912dacb6588b63b385020d5b210fb27eb8095da0 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 912dacb6588b63b385020d5b210fb27eb8095da0 -->
<!-- homu: {"type":"Approved","sha":"912dacb6588b63b385020d5b210fb27eb8095da0","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 912dacb6588b63b385020d5b210fb27eb8095da0 with merge 22f3e280103c6bfbbf90bf1ca2a650186182d32b...
<!-- homu: {"type":"BuildStarted","head_sha":"912dacb6588b63b385020d5b210fb27eb8095da0","merge_sha":"22f3e280103c6bfbbf90bf1ca2a650186182d32b"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/5710686025/job/15471252945)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/5710686025/job/15471252945","builder_name":"checks-actions"} -->

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-16`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/5710686025/job/15471220364) [(plain)](https://github.com/rust-lang-ci/rust/commit/22f3e280103c6bfbbf90bf1ca2a650186182d32b/checks/15471220364/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

```

</details>

sethp committed:
fix: lookup user by name, not id

Previously, the `id -u $LOCAL_USER_ID` check would succeed, because there was already a user mapped to the $LOCAL_USER_ID with a different name.

Looking up the entry in the database by name, however, correctly makes the `useradd` operation idempotent. An alternative fix would be to avoid duplicating an already-existing-ID, however that would require either probing for an available ID or identifying where LOCAL_USER_ID is being set and changing it to better match the ubuntu base image, neither of which is a one-line change.

sethp commented:
@rustbot label -S-waiting-on-author +S-waiting-on-review

Mark-Simulacrum commented:
@bors r+

bors commented:
:pushpin: Commit fff66a954c7664d8263a00f3ca5f021dc0ca5716 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum fff66a954c7664d8263a00f3ca5f021dc0ca5716 -->
<!-- homu: {"type":"Approved","sha":"fff66a954c7664d8263a00f3ca5f021dc0ca5716","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit fff66a954c7664d8263a00f3ca5f021dc0ca5716 with merge 9fca8e7517fd714685de8e580d8e0e08073c8e16...
<!-- homu: {"type":"BuildStarted","head_sha":"fff66a954c7664d8263a00f3ca5f021dc0ca5716","merge_sha":"9fca8e7517fd714685de8e580d8e0e08073c8e16"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/5780661002/job/15666050265)
Approved by: Mark-Simulacrum
Pushing 9fca8e7517fd714685de8e580d8e0e08073c8e16 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Mark-Simulacrum","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/5780661002/job/15666050265"},"merge_sha":"9fca8e7517fd714685de8e580d8e0e08073c8e16"} -->

rust-timer commented:
Finished benchmarking commit (9fca8e7517fd714685de8e580d8e0e08073c8e16): [comparison URL](https://perf.rust-lang.org/compare.html?start=2aae331706fee2a901766a0f0c008b263b077242&end=9fca8e7517fd714685de8e580d8e0e08073c8e16&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=2aae331706fee2a901766a0f0c008b263b077242&end=9fca8e7517fd714685de8e580d8e0e08073c8e16&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=2aae331706fee2a901766a0f0c008b263b077242&end=9fca8e7517fd714685de8e580d8e0e08073c8e16&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean | range        | count |
|:----------------------------------:|:----:|:------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -    | -            | 0     |
| Regressions ❌ <br /> (secondary)  | 2.7% | [2.5%, 2.9%] | 2     |
| Improvements ✅ <br /> (primary)   | -    | -            | 0     |
| Improvements ✅ <br /> (secondary) | -    | -            | 0     |
| All ❌✅ (primary)                 | -    | -            | 0     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=2aae331706fee2a901766a0f0c008b263b077242&end=9fca8e7517fd714685de8e580d8e0e08073c8e16&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.

### [Binary size](https://perf.rust-lang.org/compare.html?start=2aae331706fee2a901766a0f0c008b263b077242&end=9fca8e7517fd714685de8e580d8e0e08073c8e16&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: 650.283s -> 651.653s (0.21%)
